src-chrome-mv3/options.js

```diff
@@ -0,0 +1,164 @@
+
+window.addEventListener("DOMContentLoaded",
+  function () {
+
+    //GET advanced search criteria lists and populate them to drop-down
+    const subjList = document.getElementById("subject");
+    const typeList = document.getElementById("type");
+    const accList = document.getElementById("accommodation");
+    const srcList = document.getElementById("source");
+
+    fetch(`https://imgsdev.wpengine.com/json-api/subjects/`, {
+      method: 'GET',
+    }).then(response => response.json())
+      .then (json => {
+      const subjects = json.data
+
+      subjects.forEach(element => {
+        const option = document.createElement('option');
+          option.innerText = element.attributes.name;
+          option.value = element.id;
+          subjList.prepend(option);
+      });
+    })
+
+    fetch(`https://imgsdev.wpengine.com/json-api/types/`, {
+      method: 'GET',
+    }).then(response => response.json())
+      .then (json => {
+      const types = json.data
+
+      types.forEach(element => {
+        const option = document.createElement('option');
+          option.innerText = element.attributes.name;
+          option.value = element.id;
+          typeList.prepend(option);
+      });
+    })
+
+    fetch(`https://imgsdev.wpengine.com/json-api/accommodations/`, {
+      method: 'GET',
+    }).then(response => response.json())
+      .then (json => {
+      const accommodations = json.data
+
+      accommodations.forEach(element => {
+        const option = document.createElement('option');
```

Comment:
This block of code looks like a good candidate to refactor.

```javascript
function addOptions(list, target) {
  list.forEach(item => {
    const option = document.createElement('option');
    option.innerText = item.attributes.name;
    option.value = item.id;
    target.prepend(option);
  });
};

addOptions(json.data, accList);
```

Reply:
change enacted in alerts-cross-browser